### ISSUE TITLE
Combat log buff tracking issue

### ISSUE THREAD

**Issue by FurioCC** [ccshiro/cc-buglist#82]
I found another issue with combat log buff tracking (also linking for reference original report about lifebloom stack tracking weirdness http://pastebin.com/ETaNsSgG which was caused by server sometimes not firing SPELL_AURA_REFRESH that was looked into by Shiro, and seems to be working as intended now).

New issue causes breaking addons which rely on combat log for your buff tracking.

Issue is caused by server firing for unknown reason event SPELL_AURA_REMOVED thus removing buff and then in the same moment followed by SPELL_AURA_APPLIED reapplying buff (with all the stacks).
Since addon sees event of my buff being removed and does not see event of me casting for aura application, but sees aura being applied, it no longer considers me as buff owner and no longer shows it to me.

Proofs and explanations : 
https://www.dropbox.com/s/mn6lanty22s7y8m/ScrnShot_2015-01-17%2023.55.15.png?dl=0
Screen shows excerpt of filtered events from fight with Kazzak with Spellname Lifebloom and Target Nyew (tank). (Note that bug happens with other hots as well, such as Rejuvenation and Regrowth, i'm just using Lifebloom as example).

I will be referencing log lines with first field which i added, named Code serving as unique record identifier.

Log shows me rolling full stack of lifebloom on Nyew. I succesfully cast Lifebloom to refresh it (Code 22125), server sees it and fires SPELL_AURA_REFRESH (Code 22129), 4 ticks happen, i cast successfully again (Code 22218), refresh happens (Code 22227) and then suddenly server fires
SPELL_AURA_REMOVED (Code 22230) and in the same very time SPELL_AURA_APPLIED (Code 222231) and SPELL_AURA_APPLIED_DOSE (Code 22232).

After these events addons no longer consider me as buff owner, i'm unable to track it, it ticks out 7 times and normally expires with blooming lifebloom SPELL_HEAL (Code 22419) and SPELL_AURA_REMOVED (Code 22421).
In cases when i refresh Lifeboom after this weird sequence of events, but before it normally expires i still not considered as buff owner.

On another screenshot i am providing unfiltered events so you can see other events happening at that time https://www.dropbox.com/s/7qdbbvtibnl3fq8/ScrnShot_2015-01-18%2000.33.11.png?dl=0
and providing you combat log itself https://www.dropbox.com/s/v2t90xduvputzvm/WoWCombatLog_17_01.rar?dl=0

Ready to help as much as i can with any further research on issue.

**Comment by FurioCC** [comment]
Today to reproduce the issue i went to Terrokkar alone and started combat with Timber Worg,
while keeping stack of lifebloom on myself. Sometimes i let it expire on purpose to see how it behaves. Overall combat lasted about 5 minutes.

I was able to reproduce issue 2 times (with combat log showing event sequence of SPELL_AURA_REMOVED followed SPELL_AURA_APPLIED and SPELL_AURA_APPLIED_DOSE exactly at the moments when that happened).

When it happened first time i was able to recover by stopping meleeing wolf and targeting myself and refreshing lifebloom, which again gave me information about my whole stack.

Second time when it happened i kept meleeing wolf and forcefully refreshed lifebloom stack while targeting wolf and didn't get back stack info so i let it expire.

Unfortunately i do not know what causes this event sequence. Here's log of combat with wolf https://www.dropbox.com/s/xz28xew2y5r30gw/WoWCombatLog_18_01_15_wolf.rar?dl=0

I made another session about 10 minutes long and never lost stack. Combat log of that session doesn't have REMOVE and APPLY sequence. I see no point in uploading it.
